contributors/devel/cherry-picks.md

```diff
@@ -19,26 +20,55 @@ task is for backporting PRs from master to release branches.
    github.com/github/hub` assuming you have a standard golang development
    environment.
 
-## Initiate a Cherry Pick
- * Run the [cherry pick script](https://git.k8s.io/kubernetes/hack/cherry_pick_pull.sh).
+## Initiate a Cherrypick
+ * Run the [cherrypick script](https://git.k8s.io/kubernetes/hack/cherry_pick_pull.sh).
    This example applies a master branch PR #98765 to the remote branch
    `upstream/release-3.14`: `hack/cherry_pick_pull.sh upstream/release-3.14
    98765`
+   * Be aware the cherrypick script assumes you have a git remote called `upstream` that points at the Kubernetes github org.
+   It will also expect the following environment variables:
+   ```bash
+   FORK_REMOTE=<remote-of-your-fork> \
+   UPSTREAM_REMOTE=<remote-of-upstream> \
+   GITHUB_USER=<your-github-user> \
+   ./hack/cherry_pick_pull.sh <remote-of-upstream>/<release-x.xx> <master-branch-pr-number>
```

Comment:
I'm not sure if we should include all this info here. 

> ./hack/cherry_pick_pull.sh <remote-of-upstream>/<release-x.xx> <master-branch-pr-number>

This is not needed because an example already exists above.

> GITHUB_USER=<your-github-user>

This is useful because this is _really needed_. The script won't work without it. But if you don't set this env var, the script will prompt you to do that. We can include this. :+1: 

```
FORK_REMOTE=<remote-of-your-fork> \
   UPSTREAM_REMOTE=<remote-of-upstream> \
```

Looking at this, it reads to me as I _need_ to set these env vars to get the script to work. This is not true because the script will assume the default fork as "origin" and upstream as "upstream". If someone doesn't have the remotes set this way, only then would they have to set these variables.

Ref: the script has these lines:

```
UPSTREAM_REMOTE=${UPSTREAM_REMOTE:-upstream}
FORK_REMOTE=${FORK_REMOTE:-origin}
```

@neolit123 @guineveresaenger thoughts?

Reply:
> UPSTREAM_REMOTE=${UPSTREAM_REMOTE:-upstream}
FORK_REMOTE=${FORK_REMOTE:-origin}

it depends on the git workflow, i guess.

1.

the steps here suggest cloning upstream:
https://github.com/kubernetes/kubernetes/blob/master/README.md#to-start-developing-kubernetes
`git clone https://github.com/kubernetes/kubernetes`

this will set `origin` being the default remote for upstream `k/k`.

from there users can add their fork as another remote (this is what i do).
so with this workflow the `cherry_pick_pull.sh` defaults are off and the users need to rename remotes or use the env. variables.

2.

this is the official workflow however:
https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#2-clone-fork-to-local-storage
it suggests to do:
`git clone https://github.com/$user/kubernetes.git`
and them setup`upstream`, which will be compliant with the `cherry_pick_pull.sh` defaults.

the github suggested workflow is the same:
https://help.github.com/articles/configuring-a-remote-for-a-fork/

so if assume that the users always follow 2. there is no need to set the env.vars.

Reply:
whoa, I had _no idea_ there were instructions floating around that effectively set k/k as origin. We should fix that.

Reply:
`origin` as the upstream remote predates github. :)